contracts/src/l1/TokenBridge.sol

```diff
@@ -80,103 +101,109 @@ contract TokenBridge is
     }
 
-    function approveBridge(IStaticATokenLM l1Token, uint256 l2Token)
+    function approveBridge(address l1AToken, uint256 l2Token)
         external
         onlyGovernance
         onlyValidL2Address(l2Token)
     {
-        require(l1Token != IStaticATokenLM(0x0), "l1Token address cannot be 0x0");
+        require(l1AToken != address(0x0), "l1Token address cannot be 0x0");
 
-        uint256 l2Token_ = l1TokentoL2Token[l1Token];
-        require(l2Token_ == 0, "l2Token already set");
+        require(
+            aTokenData[l1AToken].l2TokenAddress == 0,
+            "l2Token already set"
+        );
 
-        require(IStaticATokenLM(l1Token).REWARD_TOKEN() == rewardToken, "L1 TOKEN CONFIGURED WITH WRONG REWARD TOKEN");
+        require(
+            IATokenWithPool(l1AToken).getIncentivesController() ==
+                incentivesController,
+            "L1 TOKEN CONFIGURED WITH WRONG INCENTIVES CONTROLLER"
+        );
 
-        l1Token.ASSET().safeApprove(address(l1Token), type(uint256).max);
-        l1Token.ATOKEN().safeApprove(address(l1Token), type(uint256).max);
+        IERC20 underlyingAsset = IERC20(
+            IATokenWithPool(l1AToken).UNDERLYING_ASSET_ADDRESS()
+        );
+        ILendingPool lendingPool = IATokenWithPool(l1AToken).POOL();
+        underlyingAsset.safeApprove(address(lendingPool), type(uint256).max);
 
-        emit LogBridgeAdded(l1Token, l2Token);
-        l1TokentoL2Token[l1Token] = l2Token;
-        approvedL1Tokens.push(l1Token);
+        aTokenData[l1AToken] = ATokenData(
+            l2Token,
+            underlyingAsset,
+            lendingPool
+        );
+        approvedL1Tokens.push(l1AToken);
+        emit LogBridgeAdded(l1AToken, l2Token);
     }
 
     function claimOrderSwap(uint256 idx1, uint256 idx2) external {
-      require(idx1 < approvedL1Tokens.length, "INDEX OUT OF RANGE");
-      require(idx2 < approvedL1Tokens.length, "INDEX OUT OF RANGE");
+        require(idx1 < approvedL1Tokens.length, "INDEX OUT OF RANGE");
+        require(idx2 < approvedL1Tokens.length, "INDEX OUT OF RANGE");
 
-      (approvedL1Tokens[idx1], approvedL1Tokens[idx2]) =
-        (approvedL1Tokens[idx2], approvedL1Tokens[idx1]);
+        (approvedL1Tokens[idx1], approvedL1Tokens[idx2]) = (
+            approvedL1Tokens[idx2],
+            approvedL1Tokens[idx1]
+        );
     }
 
     function sendMessage(
-        IStaticATokenLM l1Token,
+        address l1Token,
         address from,
         uint256 l2Recipient,
         uint256 amount,
         uint256 blockNumber,
         uint256 currentRewardsIndex
-    )
-        internal
-        onlyApprovedToken(l1Token)
-        onlyValidL2Address(l2Recipient)
-    {
-        emit LogDeposit(from, l1Token, amount, l2Recipient, blockNumber,
-                        currentRewardsIndex);
-
-        uint256 l2TokenAddress = l1TokentoL2Token[l1Token];
+    ) internal onlyValidL2Address(l2Recipient) {
```

Comment:
We are probably doing this check more than once on each code path. I think we should remove it here.